pennylane/templates/layers/strongly_entangling.py

```diff
@@ -100,23 +39,95 @@ def StronglyEntanglingLayers(weights, wires, ranges=None, imprimitive=CNOT):
 
-    Raises:
-        ValueError: if inputs do not have the correct format
+    .. UsageDetails::
+
+        **Parameter shape**
+
+        The expected shape for the weight tensor can be computed with the static method
+        :meth:`~.SimplifiedTwoDesign.shape` and used when creating randomly
+        initialised weight tensors:
+
+        .. code-block:: python
+
+            shape = StronglyEntanglingLayers.shape(n_layers=2, n_wires=2)
+            weights = np.random.random(size=shape)
+
     """
-    wires = Wires(wires)
-    repeat, ranges = _preprocess(weights, wires, ranges)
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, weights, wires, ranges=None, imprimitive=None, do_queue=True):
+
+        shape = qml.math.shape(weights)
+        self.n_layers = shape[0]
+
+        if len(shape) != 3:
+            raise ValueError(f"Weights tensor must be 3-dimensional; got shape {shape}")
+
+        if shape[1] != len(wires):
+            raise ValueError(
+                f"Weights tensor must have second dimension of length {len(wires)}; got {shape[1]}"
+            )
+
+        if shape[2] != 3:
+            raise ValueError(
+                f"Weights tensor must have third dimension of length 3; got {shape[2]}"
+            )
+
+        if ranges is None:
+            if len(wires) > 1:
+                # tile ranges with iterations of range(1, n_wires)
+                self.ranges = [(l % (len(wires) - 1)) + 1 for l in range(self.n_layers)]
+            else:
+                self.ranges = [0] * self.n_layers
+
+        self.imprimitive = imprimitive or qml.CNOT
+
+        super().__init__(weights, wires=wires, do_queue=do_queue)
+
+    def expand(self):
+
+        with qml.tape.QuantumTape() as tape:
+
+            for l in range(self.n_layers):
+
+                for i in range(len(self.wires)):
+                    qml.Rot(
+                        self.parameters[0][l, i, 0],
+                        self.parameters[0][l, i, 1],
+                        self.parameters[0][l, i, 2],
+                        wires=self.wires[i],
+                    )
+
+                if len(self.wires) > 1:
+                    for i in range(len(self.wires)):
+                        act_on = self.wires.subset([i, i + self.ranges[l]], periodic_boundary=True)
+                        self.imprimitive(wires=act_on)
+
+        return tape
+
+    @staticmethod
+    def shape(n_layers, n_wires):
+        r"""Returns the expected shape of the weights tensor.
+
+        Args:
+            n_layers (int): number of layers
+            n_wires (int): number of wires
 
-    for l in range(repeat):
+        Returns:
+            tuple[int]: shape
+        """
 
-        strongly_entangling_layer(
-            weights=weights[l], wires=wires, r=ranges[l], imprimitive=imprimitive
-        )
+        return n_layers, n_wires, 3
```

Comment:
```suggestion
        return (n_layers, n_wires, 3)
```
Doesn't matter at all but makes it more clear to me it should be treated as a tuple and not 3 separate return args.

Reply:
Codefactor complains about the unnecessary brakets though...?